src/platform/GenericOTARequestorDriver.h

```diff
@@ -42,23 +44,45 @@ class GenericOTARequestorDriver : public OTARequestorDriver
         mImageProcessor = processor;
     }
 
+    // Set the timeout (in seconds) for the Default Provider timer
+    void SetDefaultProvidersTimeoutSec(uint32_t timeout) { mDefaultProvidersTimeoutSec = timeout; }
+
+    //// Virtual methods from OTARequestorDriver
     bool CanConsent() override;
     uint16_t GetMaxDownloadBlockSize() override;
-
     void HandleError(UpdateFailureState state, CHIP_ERROR error) override;
+    void HandleIdleState() override;
     void UpdateAvailable(const UpdateDescription & update, System::Clock::Seconds32 delay) override;
     void UpdateNotFound(UpdateNotFoundReason reason, System::Clock::Seconds32 delay) override;
     void UpdateDownloaded() override;
     void UpdateConfirmed(System::Clock::Seconds32 delay) override;
     void UpdateSuspended(System::Clock::Seconds32 delay) override;
     void UpdateDiscontinued() override;
     void UpdateCancelled() override;
+    void ScheduleDelayedAction(UpdateFailureState state, System::Clock::Seconds32 delay, System::TimerCompleteCallback action, void * aAppState) override;
+    void CancelDelayedAction(System::TimerCompleteCallback action, void * aAppState) override;
+    void ProcessAnnounceOTAProviders(const ProviderLocationType &providerLocation, 
+                                        app::Clusters::OtaSoftwareUpdateRequestor::OTAAnnouncementReason announcementReason) override;
+    void DriverSendQuery() override;
+    // Determines the next available Provider location and sets it in the OTARequestor
+    void DetermineAndSetProviderLocation() override;
+   
+    //// Regular methods
+    void StartDefaultProvidersTimer();
+    void StopDefaultProvidersTimer();
+    void DefaultProviderTimerHandler(System::Layer * systemLayer, void * appState);
+ 
+private:
+    // Returns the next available Provider location
+    bool DetermineProviderLocation(app::Clusters::OtaSoftwareUpdateRequestor::Structs::ProviderLocation::Type & providerLocation);
 
-protected:
-    void ScheduleDelayedAction(UpdateFailureState state, System::Clock::Seconds32 delay, System::TimerCompleteCallback action);
-
-    OTARequestorInterface * mRequestor           = nullptr;
+    OTARequestorInterface      * mRequestor           = nullptr;
     OTAImageProcessorInterface * mImageProcessor = nullptr;
+    uint32_t mOtaStartDelayMs                 = 0;
+    uint32_t mDefaultProvidersTimeoutSec      = 86400;  // Timeout for the Default Provider timer
```

Comment:
Is this timer used explicitly for periodic queries? Or is it used for something else as well?

Reply:
Just the periodic queries